mizar/networkpolicy/networkpolicy_util.py

```diff
@@ -30,26 +30,76 @@
 
 
 class NetworkPolicyUtil:
-    def handle_networkpolicy_create_update(self, name, pod_label_dict, policy_types):
+    def update_networkpolicy_endpoints_mapping_and_return_affected_endpoint_names(self, policy_name, pod_label_dict):
+        affected_endpoint_names = set()
+
+        endpoint_name_list = self.retrieve_endpoints_for_networkpolicy(policy_name, pod_label_dict)
+
+        if policy_name not in networkpolicy_opr.store.networkpolicy_endpoints_store:
+            networkpolicy_opr.store.networkpolicy_endpoints_store[policy_name] = set()
+
+        for endpoint_name in networkpolicy_opr.store.networkpolicy_endpoints_store[policy_name]:
+            if endpoint_name in endpoint_name_list:
+                endpoint_name_list.remove(endpoint_name)
+            else:                
+                affected_endpoint_names.add(endpoint_name)
+                ep = networkpolicy_opr.store.get_ep(endpoint_name)
+                if ep is not None:
+                    ep.remove_ingress_networkpolicy(policy_name)
+                    ep.remove_egress_networkpolicy(policy_name)
+
+        for endpoint_name in affected_endpoint_names:
+            networkpolicy_opr.store.networkpolicy_endpoints_store[policy_name].remove(endpoint_name)
+
+        for endpoint_name in endpoint_name_list:
+            networkpolicy_opr.store.add_networkpolicy_endpoint(policy_name, endpoint_name)
+            affected_endpoint_names.add(endpoint_name)
+
+        if len(networkpolicy_opr.store.networkpolicy_endpoints_store[policy_name]) == 0:
+            networkpolicy_opr.store.networkpolicy_endpoints_store.pop(policy_name)
+
+        return affected_endpoint_names
+
+    def retrieve_endpoints_for_networkpolicy(self, policy_name, pod_label_dict):
+        endpoint_name_list = set()
+
+        if pod_label_dict is None:
+            return endpoint_name_list
+
         pods = kube_list_pods_by_labels(networkpolicy_opr.core_api, pod_label_dict)
         if pods is None:
-            return
+            return endpoint_name_list
+        
+        for pod in pods.items:
+            pod_name = pod.metadata.name
+            eps = endpoint_opr.store.get_eps_in_pod(pod_name)
+            for ep in eps.values():
+                endpoint_name_list.add(ep.name)
```

Comment:
not sure if pods in different namespace could be handled properly. Is the ep name has namespace as part of its name?

Reply:
This is known issue: mizar doesn't handle namespace. Which means when the objects are in different namespaces, mizar just ignores namespace but only use name property. When there is same name objects in different namespaces, the system result is unexpected.
I created issue https://github.com/CentaurusInfra/mizar/issues/365 for tracking.

Reply:
OK got it. Thanks for opening the issue tracking it.